docs/data/sdks/typescript-ampli.md

```diff
@@ -0,0 +1,346 @@
+      role: 'admin'
+    }, {
+      deviceId: 'my-device-id'
+    });
+    ```
+
+=== "JavaScript"
+
+    ```js
+    ampli.identify('user-id', {
+      role: 'admin'
+    }, {
+      deviceId: 'my-device-id'
+    });
+    ```
+
+### Group
+
+!!! note
+    This feature is available for Growth customers who have purchased the [Accounts add-on](https://help.amplitude.com/hc/en-us/articles/115001765532).
+
+Call `setGroup()` to associate a user with their group (for example, their department or company). The `setGroup()` function accepts a required `groupType`, and `groupName`.
+
+=== "TypeScript"
+
+    ```js
+    ampli.setGroup('groupType', 'groupName');
+    ```
+
+=== "JavaScript"
+
+    ```js
+    ampli.setGroup('groupType', 'groupName');
+    ```
+
+--8<-- "includes/groups-intro-paragraph.md"
+
+ Setting a group also sets the 'groupType:groupName' as a user property, and overwrites any existing groupName value set for that user's groupType, and the corresponding user property value. groupType is a string, and groupName can be either a string or an array of strings to indicate that a user is in multiple groups. For example, if Joe is in 'orgId' '10' and '20', then the `groupName` is '[10, 20]').
+
+ Your code might look like this:
+
+=== "TypeScript"
+
+    ```js
+    ampli.setGroup('orgId', ['10', '20']);
+    ```
+
+=== "JavaScript"
+
+    ```js
+    ampli.setGroup('orgId', ['10', '20']);
+    ```
+
+### Track
+
+To track an event, call the event's corresponding function. Every event in your tracking plan gets its own function in the Ampli SDK. The call is structured like this:
+
+=== "TypeScript"
+
+    ```js
+    ampli.eventName(properties: EventNameProperties, options: EventOptions, extra: MiddlewareExtra)
+    ```
+
+=== "JavaScript"
+
+    ```js
+    ampli.eventName(properties: EventNameProperties, options: EventOptions, extra: MiddlewareExtra)
+    ```
+
+The `properties` argument passes event properties.
+
+The `options` argument allows you to pass to pass [Amplitude fields](https://developers.amplitude.com/docs/http-api-v2#properties-1), like `price`, `quanity` and `revenue`.
+
+The `extra` argument lets you pass data to middleware.
+
+For example, in the code snippet below, your tracking plan contains an event called `songPlayed`. The event is defined with two required properties: `songId` and `songFavorited`.
+ The property type for `songId` is string, and `songFavorited` is a boolean.
+
+The event has an Amplitude field defined: `deviceId`. Learn more about Amplitude fields [here](https://developers.amplitude.com/docs/http-api-v2#properties-1).
+ The event has one MiddlewareExtra defined: `myMiddleware`. Learn more about [Middleware](#middleware).
+
+=== "TypeScript"
+
+    ```js
+    ampli.songPlayed( {
+      songId: 'songId', // string,
+      songFavorited: true, // boolean
+    }, {
+      deviceId: 'a-device-id',
+    }, {
+      myMiddleware: { myMiddlewareProp: "value to send to middleware" }
```

Comment:
No middleware anymore in ampli v2 and the second parameter is EventOptions.

Reply:
+1